parser_test.go

```diff
@@ -1333,3 +1338,65 @@ func TestUTF8NonPrintableArtifacts(t *testing.T) {
 		})
 	}
 }
+
+// According to the JSON spec: https://datatracker.ietf.org/doc/html/rfc8259#section-8.1
+// > "JSON text exchanged between systems that are not part of a closed ecosystem MUST be encoded using UTF-8"
+// However, go strings are not *enforced* to be UTF-8. They are represented as such but anything can reside within them.
+//
+// This test ensure that in case of non-utf-8 bytes, the fastjson library can produce valid string and will replace
+// the invalid bytes with the Unicode replacement character. (U+FFFD)
+// This is what the encoding/json does as well.
+func TestNonUTF8(t *testing.T) {
```

Comment:
Awesome work on this test, especially your comments 💯

Reply:
Thanks, you helped a ton with this ❤️